src/main/java/bio/terra/service/dataset/flight/ingest/IngestUtils.java

```diff
@@ -133,4 +137,20 @@ public static PdaoLoadStatistics getIngestStatistics(FlightContext context) {
     FlightMap workingMap = context.getWorkingMap();
     return workingMap.get(IngestMapKeys.INGEST_STATISTICS, PdaoLoadStatistics.class);
   }
+
+  public static Predicate<FlightContext> noFilesToIngestPredicate() {
+    return flightContext -> {
+      if (flightContext.getFlightClassName().equals(DatasetIngestFlight.class.getName())
```

Comment:
Depending on a class name seems fragile to me, is there no better way to determine if this predicate should run? What about the presence of a map key?

Reply:
This is a holdover from development. Its been removed since this is always called from DatasetIngestFlight.